app/src/test/java/org/greenstand/android/TreeTracker/messages/individualmessagelist/IndividualMessageListViewModelTest.kt

```diff
@@ -0,0 +1,74 @@
+package org.greenstand.android.TreeTracker.messages.individualmessagelist
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.runBlocking
+import org.greenstand.android.TreeTracker.MainCoroutineRule
+import org.greenstand.android.TreeTracker.messages.individualmeassagelist.IndividualMessageListViewModel
+import org.greenstand.android.TreeTracker.models.UserRepo
+import org.greenstand.android.TreeTracker.models.messages.MessagesRepo
+import org.greenstand.android.TreeTracker.utils.FakeFileGenerator
+import org.greenstand.android.TreeTracker.utils.getOrAwaitValueTest
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class IndividualMessageListViewModelTest {
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+    private val userId = 2L
+    private val userRepo = mockk<UserRepo>(relaxed = true)
+    private val messagesRepo = mockk<MessagesRepo>(relaxed = true)
+    private lateinit var individualMessageListViewModel: IndividualMessageListViewModel
+
+    @Before
+    fun setup(){
+        coEvery { userRepo.getUser(any()) } returns FakeFileGenerator.fakeUsers.first()
+        coEvery { messagesRepo.getMessageFlow(any()) } returns flowOf(FakeFileGenerator.messages)
+        individualMessageListViewModel = IndividualMessageListViewModel(userId, userRepo, messagesRepo)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun `verify user repo gets the correct user`() = runBlocking {
+       coVerify { userRepo.getUser(userId) }
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun `verify message repo gets the correct message flow`() = runBlocking {
+        val currentUser = userRepo.getUser(userId)
+        val wallet = currentUser!!.wallet
+        coVerify { messagesRepo.getMessageFlow(wallet) }
+    }

```

Comment:
This method does not test the testSubject, and therefor should not be here.

Instead we just want to setup the mocked data and then check that the testSubjects state is correct.